### ISSUE TITLE
SteamVR setting to force the handedness of a given psmove controller

### ISSUE THREAD

**Issue by alerrandro** [cboulay/PSMoveService#156]
How I can work with only one ps move controller , i always get second controller on steam vr and its big mess! i tried configure on steam vr settings or appdata /trackmanangerconfig but didn't work!

**Comment by HipsterSloth** [comment]
Do you have more than one controller connected and you only want one to show up? Or you are saying you only have one controller connected and you are seeing two controllers in steamvr.

**Comment by alerrandro** [comment]
i have just one controller connected, but never mind i just change usb connection and recalibrate and works and steam vr switch position of active controller

**Comment by cboulay** [comment]
@alerrandro My interpretation of your questions is that your one controller showed up in steamvr but it showed up as (e.g.) the 'left' controller when you wanted it to be the 'right'? Maybe the reason it 'worked' after you changed the USB connection is because PSMoveService remembered another connection so the new one showed up as controller1 (instead of 0), and was assigned 'right' whereas the not-real controller0 was 'left'?

If my interpretation is correct then I think it might be possible to provide a way to set the handedness of a single controller, but I don't think we have that capability right now and I'm not sure how that would work at runtime. @HipsterSloth can comment further.

That being said, this is pretty low priority for us. While it's nice that people use PSMoveService with SteamVR, and @HipsterSloth did a great job with the driver, it's a bit of a fool's errand to get PSMove controllers, especially just 1 controller, to provide all the functionality of a pair of Vive controllers for games designed specifically for 2 Vive controllers.

Please let me know whether I understood you correctly or if you meant something else entirely.

**Comment by alerrandro** [comment]
yeah you get it !!! im already bought a ps move controller

**Comment by HipsterSloth** [comment]
Yeah setting the handedness of the controller is straightforward. At one point I actually was manually assigning a hand based on the controller order in an attempt to fix an issue, see 7c9a94685d46dd7a4938aedc46c4f1d6e9d24c86. That has since been ripped back out as I learned that it was the wrong way to go about solving that particular problem. But it would be straight forward to add that back in as a configurable option in the steamvr.vrsettings file (similar to how we set button mappings for steamvr). I don't think it should be a setting in the config tool, because PSMove controllers don't really have a handedness to them.

I'll update the bug title to reflect this request.
